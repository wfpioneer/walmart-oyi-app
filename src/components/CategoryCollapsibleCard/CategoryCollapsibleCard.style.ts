import { StyleSheet } from 'react-native';
import COLOR from '../../themes/Color';

export default StyleSheet.create({
  categoryList: {
    flex: 1
  },
  categoryFilterCard: {
    alignItems: 'center',
    justifyContent: 'flex-start',
    flexDirection: 'row',
    flex: 1,
    height: 40
  },
  categoryFilterText: {
    width: '90%'
  },
  menuCard: {
    alignItems: 'center',
    flexDirection: 'row',
    paddingVertical: 15,
    paddingHorizontal: 10,
    borderBottomWidth: 1,
    borderBottomColor: COLOR.GREY_300
  },
  selectionView: {
    alignItems: 'center',
    justifyContent: 'center',
    width: '10%'
  }
});
