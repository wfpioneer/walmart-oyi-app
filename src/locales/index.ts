import * as RNLocalize from 'react-native-localize';
import I18n from 'i18n-js';
// @ts-ignore
import moment from 'moment';
import 'moment/locale/es';
import en from './en';
import es from './es';
import zh from './zh';

export const setI18nConfig: () => void = () => {
  // fallback if no available language fits
  const fallback = { languageTag: 'en', isRTL: false };

  const { languageTag } = RNLocalize.findBestAvailableLanguage(['en', 'es', 'zh'])
    || fallback;

  // set i18n-js config
  I18n.translations = { en, es, zh };
  I18n.locale = languageTag;

  // the following is to set up moment to work with spanish localization
  moment.locale(languageTag);
};

export const setLanguage = (language: string) => {
  if (language !== 'en' && language !== 'es' && language !== 'zh') {
    throw new Error('Unsupported language');
  }

  I18n.locale = language;
  moment.locale(language);
};

export const strings: (
  tag: string,
  options?: I18n.TranslateOptions | undefined,
) => string = (tag: string, options: I18n.TranslateOptions | undefined = {}) => I18n.t(tag, options);

export const currencies: (
  value: number,
  options?: I18n.ToCurrencyOptions | undefined
) => string = (value: number, options: I18n.ToCurrencyOptions | undefined = {}) => I18n.toCurrency(value, options);

export const numbers: (
  value: number,
  options?: I18n.ToNumberOptions | undefined
) => string = (value: number, options: I18n.ToNumberOptions | undefined = {}) => I18n.toNumber(value, options);
