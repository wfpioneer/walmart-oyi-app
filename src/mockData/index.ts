
import itemDetails from './getItemDetails';


export const getMockItemDetails = (upc: string) => {
  // @ts-ignore
  return itemDetails[upc] || {};
};
