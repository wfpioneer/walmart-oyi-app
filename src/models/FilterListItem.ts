interface FilterListItem {
    value: string;
    display: string;
    selected: boolean;
  }

export default FilterListItem;
