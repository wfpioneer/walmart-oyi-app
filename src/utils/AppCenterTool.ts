import AppCenter from 'appcenter';
import AppCenterCrashes from 'appcenter-crashes';
import AppCenterAnalytics from 'appcenter-analytics';
import store from '../state';

export const initialize = (): void => {
  if (!__DEV__) {
    AppCenter.setLogLevel(AppCenter.LogLevel.VERBOSE);
    AppCenterCrashes.notifyUserConfirmation(AppCenterCrashes.UserConfirmation.ALWAYS_SEND);
    AppCenterCrashes.setEnabled(true);
    AppCenterAnalytics.setEnabled(true);
  }
};

export const trackEvent = (eventName: string, params: any = {}): void => {
  const userState = store.getState().User;
  // Removes undefined values from the event properties
  const cleanedParams = Object.fromEntries(Object.entries(params).filter(([k, v]) => v !== undefined));
  const submitParams = {
    ...cleanedParams,
    userId: userState.userId,
    clubId: userState.siteId.toString(),
    country: userState.countryCode
  };
  AppCenterAnalytics.trackEvent(eventName, submitParams);
};

export const setUserId = (userId: string): void => {
  AppCenter.setUserId(userId);
};
