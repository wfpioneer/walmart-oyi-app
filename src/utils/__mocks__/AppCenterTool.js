export default {
  initialize: jest.fn(),
  trackEvent: jest.fn(() => Promise.resolve()),
  setUserId: jest.fn(() => Promise.resolve())
};
