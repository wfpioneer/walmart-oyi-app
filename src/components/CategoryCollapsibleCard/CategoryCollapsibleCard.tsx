import React from 'react';
import {
  FlatList,
  Text,
  TouchableOpacity,
  View
} from 'react-native';
import { Dispatch } from 'redux';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import { FilteredCategory } from '../../models/FilterListItem';
import { MenuCard } from '../../screens/Worklist/FilterMenu/FilterMenu';
import { strings } from '../../locales';
import { trackEvent } from '../../utils/AppCenterTool';
import COLOR from '../../themes/Color';
import styles from './CategoryCollapsibleCard.style';

export const renderCategoryFilterCard = (
  item: FilteredCategory,
  dispatch: Dispatch<any>,
  filterCategories: string[],
  source: string,
  updateFilterCategories: (categories: string[]) => void
): JSX.Element => {
  const onItemPress = () => {
    if (item.selected) {
      filterCategories.splice(
        filterCategories.indexOf(`${item.catgNbr} - ${item.catgName}`),
        1
      );
      trackEvent(`${source}_update_filter_categories`, {
        categories: JSON.stringify(filterCategories)
      });
      return dispatch(updateFilterCategories(filterCategories));
    }

    const replacementFilter = filterCategories;
    replacementFilter.push(`${item.catgNbr} - ${item.catgName}`);
    return dispatch(updateFilterCategories(replacementFilter));
  };
  return (
    <TouchableOpacity
      testID="category button"
      style={styles.categoryFilterCard}
      onPress={onItemPress}
    >
      <View style={styles.selectionView}>
        {item.selected ? (
          <MaterialCommunityIcons
            name="checkbox-marked-outline"
            size={15}
            color={COLOR.MAIN_THEME_COLOR}
          />
        ) : (
          <MaterialCommunityIcons
            name="checkbox-blank-outline"
            size={15}
            color={COLOR.MAIN_THEME_COLOR}
          />
        )}
      </View>
      <Text style={styles.categoryFilterText} numberOfLines={2}>
        {`${item.catgNbr} - ${item.catgName} `}
      </Text>
    </TouchableOpacity>
  );
};

export const RenderCategoryCollapsibleCard = (props: {
    categoryMap: FilteredCategory[];
    categoryOpen: boolean;
    filterCategories: string[];
    dispatch: Dispatch<any>;
    source: string;
    toggleCategories: (open: boolean) => void;
    updateFilterCatgories: (categories: string[]) => void
  }): JSX.Element => {
  const {
    categoryMap, categoryOpen, filterCategories,
    dispatch, source,
    toggleCategories, updateFilterCatgories
  } = props;
  const categoryNumberMap = categoryMap.map(
    (item: FilteredCategory) => item.catgNbr
  );
  const categoryNumberSet = new Set();
  categoryNumberMap.forEach((item: any) => categoryNumberSet.add(item));
  const filteredCategories = categoryMap.filter((item: any) => {
    if (categoryNumberSet.has(item.catgNbr)) {
      categoryNumberSet.delete(item.catgNbr);
      return true;
    }
    return false;
  });

  let categorySubtext = '';
  if (filterCategories.length === 0) {
    categorySubtext = strings('WORKLIST.ALL');
  } else {
    categorySubtext = `${filterCategories.length} ${strings(
      'GENERICS.SELECTED'
    )}`;
  }

  return (
    <>
      <TouchableOpacity
        style={styles.menuCard}
        onPress={() => {
          dispatch(toggleCategories(!categoryOpen));
        }}
      >
        <MenuCard
          title={strings('WORKLIST.CATEGORY')}
          subtext={categorySubtext}
          opened={categoryOpen}
        />
      </TouchableOpacity>
      {categoryOpen && (
      <FlatList
        data={filteredCategories}
        renderItem={({ item }) => renderCategoryFilterCard(
          item,
          dispatch,
          filterCategories,
          source,
          updateFilterCatgories
        )}
        style={styles.categoryList}
        keyExtractor={(item: any) => item.catgNbr.toString()}
      />
      )}
    </>
  );
};
