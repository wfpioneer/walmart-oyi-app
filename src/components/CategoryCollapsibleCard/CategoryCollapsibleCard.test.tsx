import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import { FilteredCategory } from '../../models/FilterListItem';
import { strings } from '../../locales';
import { mockCategoryMap } from '../../mockData/mockWorkList';
import { RenderCategoryCollapsibleCard, renderCategoryFilterCard } from './CategoryCollapsibleCard';
import { trackEvent } from '../../utils/AppCenterTool';

const mockDispatch = jest.fn();
const mockFilterCategories: string[] = [
  '3 - OFFICE SUPPLIES',
  '7 - TOYS',
  '8 - PETS',
  '10 - AUTOMOTIVES',
  '12 - WINE'
];
const mockItem: FilteredCategory[] = [
  { catgNbr: 7, catgName: 'TOYS', selected: false },
  { catgNbr: 12, catgName: 'WINE', selected: true }
];
const mockUpdateCats = jest.fn();
describe('Category collapsible card for filtration render tests', () => {
  it('Test the renderCategoryCollapsibleCard and calls dispatch()', () => {
    const { toJSON, getByText } = render(
      <RenderCategoryCollapsibleCard
        categoryMap={mockCategoryMap}
        categoryOpen={false}
        filterCategories={mockFilterCategories}
        source="worklist"
        toggleCategories={(catOpen: boolean) => mockDispatch(catOpen)}
        updateFilterCatgories={(updateFilterCats: string[]) => mockDispatch(updateFilterCats)}
      />
    );
    const menuButton = getByText(strings('WORKLIST.CATEGORY'));
    fireEvent.press(menuButton);
    expect(mockDispatch).toBeCalledTimes(1);
    expect(toJSON()).toMatchSnapshot();
  });

  it('Test renders the renderCategoryCollapsibleCard and filteredCategories FlatList ', () => {
    // You cannot use queries if the component contains a FlatList and isn't a PureComponent
    const { toJSON, getByText } = render(
      <RenderCategoryCollapsibleCard
        categoryMap={mockCategoryMap}
        categoryOpen={true}
        filterCategories={mockFilterCategories}
        source="worklist"
        toggleCategories={(catOpen: boolean) => mockDispatch(catOpen)}
        updateFilterCatgories={(updateFilterCats: string[]) => mockDispatch(updateFilterCats)}
      />
    );
    const categoryButton = getByText(strings('WORKLIST.CATEGORY'));
    fireEvent.press(categoryButton);
    expect(toJSON()).toMatchSnapshot();
  });

  it('Tests renderCategoryFilterCard component and calls onItemPress without an item selected', async () => {
    const { toJSON, findByTestId } = render(
      renderCategoryFilterCard(mockItem[0], mockFilterCategories, 'worklist', mockUpdateCats)
    );
    const catButton = findByTestId('category button');
    fireEvent.press(await catButton);
    expect(mockUpdateCats).toBeCalledTimes(1);

    expect(toJSON()).toMatchSnapshot();
  });

  // This could be tested by re-rendering using render.update(), but we want to ensure that the correct Icon is rendered
  it('Tests renderCategoryFilterCard component and calls onItemPress with an item selected', async () => {
    const { toJSON, findByTestId } = render(
      renderCategoryFilterCard(mockItem[1], mockFilterCategories, 'worklist', mockUpdateCats)
    );
    const catButton = findByTestId('category button');
    fireEvent.press(await catButton);
    expect(trackEvent).toBeCalledTimes(1);
    expect(mockUpdateCats).toBeCalledTimes(1);

    expect(toJSON()).toMatchSnapshot();
  });
});
