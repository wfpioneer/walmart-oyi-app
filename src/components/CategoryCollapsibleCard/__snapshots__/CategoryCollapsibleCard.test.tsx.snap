// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category collapsible card for filtration render tests Test renders the renderCategoryCollapsibleCard and filteredCategories FlatList  1`] = `
[
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#e0e0e0",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 15,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text>
        [missing "en.WORKLIST.CATEGORY" translation]
      </Text>
      <Text
        style={
          {
            "color": "#9e9e9e",
            "fontSize": 12,
          }
        }
      >
        5 [missing "en.GENERICS.SELECTED" translation]
      </Text>
    </View>
    <View
      style={
        {
          "flex": 0.1,
        }
      }
    >
      <Icon
        color="#616161"
        name="keyboard-arrow-up"
        size={25}
      />
    </View>
  </View>,
  <RCTScrollView
    data={
      [
        {
          "catgName": "WINE",
          "catgNbr": 19,
          "selected": false,
        },
        {
          "catgName": "PHARMACY RX",
          "catgNbr": 87,
          "selected": false,
        },
        {
          "catgName": "FRESH BAKERY",
          "catgNbr": 88,
          "selected": false,
        },
        {
          "catgName": "FOODSERVICE",
          "catgNbr": 93,
          "selected": false,
        },
        {
          "catgName": "ELECTRONICS",
          "catgNbr": 99,
          "selected": false,
        },
        {
          "catgName": "ELECTRONICS",
          "catgNbr": 199,
          "selected": false,
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    style={
      {
        "flex": 1,
      }
    }
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "flex-start",
              "opacity": 1,
            }
          }
          testID="category button"
        >
          <View
            style={
              {
                "alignItems": "center",
                "justifyContent": "center",
                "width": "10%",
              }
            }
          >
            <Icon
              color="#0071e9"
              name="checkbox-blank-outline"
              size={15}
            />
          </View>
          <Text
            numberOfLines={2}
            style={
              {
                "width": "90%",
              }
            }
          >
            19 - WINE 
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "flex-start",
              "opacity": 1,
            }
          }
          testID="category button"
        >
          <View
            style={
              {
                "alignItems": "center",
                "justifyContent": "center",
                "width": "10%",
              }
            }
          >
            <Icon
              color="#0071e9"
              name="checkbox-blank-outline"
              size={15}
            />
          </View>
          <Text
            numberOfLines={2}
            style={
              {
                "width": "90%",
              }
            }
          >
            87 - PHARMACY RX 
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "flex-start",
              "opacity": 1,
            }
          }
          testID="category button"
        >
          <View
            style={
              {
                "alignItems": "center",
                "justifyContent": "center",
                "width": "10%",
              }
            }
          >
            <Icon
              color="#0071e9"
              name="checkbox-blank-outline"
              size={15}
            />
          </View>
          <Text
            numberOfLines={2}
            style={
              {
                "width": "90%",
              }
            }
          >
            88 - FRESH BAKERY 
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "flex-start",
              "opacity": 1,
            }
          }
          testID="category button"
        >
          <View
            style={
              {
                "alignItems": "center",
                "justifyContent": "center",
                "width": "10%",
              }
            }
          >
            <Icon
              color="#0071e9"
              name="checkbox-blank-outline"
              size={15}
            />
          </View>
          <Text
            numberOfLines={2}
            style={
              {
                "width": "90%",
              }
            }
          >
            93 - FOODSERVICE 
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "flex-start",
              "opacity": 1,
            }
          }
          testID="category button"
        >
          <View
            style={
              {
                "alignItems": "center",
                "justifyContent": "center",
                "width": "10%",
              }
            }
          >
            <Icon
              color="#0071e9"
              name="checkbox-blank-outline"
              size={15}
            />
          </View>
          <Text
            numberOfLines={2}
            style={
              {
                "width": "90%",
              }
            }
          >
            99 - ELECTRONICS 
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "flex-start",
              "opacity": 1,
            }
          }
          testID="category button"
        >
          <View
            style={
              {
                "alignItems": "center",
                "justifyContent": "center",
                "width": "10%",
              }
            }
          >
            <Icon
              color="#0071e9"
              name="checkbox-blank-outline"
              size={15}
            />
          </View>
          <Text
            numberOfLines={2}
            style={
              {
                "width": "90%",
              }
            }
          >
            199 - ELECTRONICS 
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>,
]
`;

exports[`Category collapsible card for filtration render tests Test the renderCategoryCollapsibleCard and calls dispatch() 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderBottomColor": "#e0e0e0",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "opacity": 1,
      "paddingHorizontal": 10,
      "paddingVertical": 15,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text>
      [missing "en.WORKLIST.CATEGORY" translation]
    </Text>
    <Text
      style={
        {
          "color": "#9e9e9e",
          "fontSize": 12,
        }
      }
    >
      5 [missing "en.GENERICS.SELECTED" translation]
    </Text>
  </View>
  <View
    style={
      {
        "flex": 0.1,
      }
    }
  >
    <Icon
      color="#616161"
      name="keyboard-arrow-down"
      size={25}
    />
  </View>
</View>
`;

exports[`Category collapsible card for filtration render tests Tests renderCategoryFilterCard component and calls onItemPress with an item selected 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "height": 40,
      "justifyContent": "flex-start",
      "opacity": 1,
    }
  }
  testID="category button"
>
  <View
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "width": "10%",
      }
    }
  >
    <Icon
      color="#0071e9"
      name="checkbox-marked-outline"
      size={15}
    />
  </View>
  <Text
    numberOfLines={2}
    style={
      {
        "width": "90%",
      }
    }
  >
    12 - WINE 
  </Text>
</View>
`;

exports[`Category collapsible card for filtration render tests Tests renderCategoryFilterCard component and calls onItemPress without an item selected 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "height": 40,
      "justifyContent": "flex-start",
      "opacity": 1,
    }
  }
  testID="category button"
>
  <View
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "width": "10%",
      }
    }
  >
    <Icon
      color="#0071e9"
      name="checkbox-blank-outline"
      size={15}
    />
  </View>
  <Text
    numberOfLines={2}
    style={
      {
        "width": "90%",
      }
    }
  >
    7 - TOYS 
  </Text>
</View>
`;
